src/utils.ts

```diff
@@ -60,3 +60,21 @@ export function getSplicePosition(
   }
   return splicePosition;
 }
+
+export function toArrayBuffer(
+  arr: Uint8Array | ArrayBuffer | DataView | object
+): ArrayBufferLike {
+  if (arr instanceof Uint8Array) {
+    return arr.buffer.slice(arr.byteOffset, arr.byteOffset + arr.byteLength);
+  }
+  if (arr instanceof DataView) {
+    return arr.buffer.slice(arr.byteOffset, arr.byteOffset + arr.byteLength);
+  }
+  if (arr instanceof ArrayBuffer) {
+    return arr;
+  }
+  // If we've made it this far, we probably have a
+  // parsed JSON object
+  const encoder = new TextEncoder();
+  return encoder.encode(JSON.stringify(arr)).buffer;
```

Comment:
I'm not super excited about this bit here, but we need some kind of handling for when the incoming data was parsed as JSON (via the `application/json` content-type), and this seemed to be the most reasonable place.